libamqpprox/amqpprox_session.h

```diff
@@ -78,18 +79,20 @@ class Session : public std::enable_shared_from_this<Session> {
     TimePoint                d_egressStartedAt;
     std::vector<boost::asio::ip::tcp::endpoint> d_resolvedEndpoints;
     uint32_t                                    d_resolvedEndpointsIndex;
+    std::shared_ptr<AuthInterceptInterface>     d_authIntercept;
 
   public:
     // CREATORS
-    Session(boost::asio::io_service &              ioservice,
-            MaybeSecureSocketAdaptor &&            serverSocket,
-            MaybeSecureSocketAdaptor &&            clientSocket,
-            ConnectionSelector *                   connectionSelector,
-            EventSource *                          eventSource,
-            BufferPool *                           bufferPool,
-            DNSResolver *                          dnsResolver,
-            const std::shared_ptr<HostnameMapper> &hostnameMapper,
-            std::string_view                       localHostname);
+    Session(boost::asio::io_service &                      ioservice,
```

Comment:
maybe its time to add a `SessionConfig` class?

Reply:
Discussed about this offline.